pjsip/src/pjsua-lib/pjsua_core.c

```diff
@@ -2033,6 +2033,22 @@ PJ_DEF(pj_status_t) pjsua_destroy2(unsigned flags)
 		pjsua_var.acc[i].pool = NULL;
 	    }
 	}
+
+	/* Destroy pool in the wav player object */
+        for (i = 0; i < (int)PJ_ARRAY_SIZE(pjsua_var.player); ++i) {
+	    if (pjsua_var.player[i].pool != NULL) {
+		pj_pool_release(pjsua_var.player[i].pool);
+		pjsua_var.player[i].pool = NULL;
+	    }
+        }
+
+	/* Destroy pool in the wav writer object */
+        for (i = 0; i < (int)PJ_ARRAY_SIZE(pjsua_var.recorder); ++i) {
+	    if (pjsua_var.recorder[i].pool != NULL) {
+		pj_pool_release(pjsua_var.recorder[i].pool);
+		pjsua_var.recorder[i].pool = NULL;
+	    }
+        }
```

Comment:
Pool will automatically be released when the factory is destroyed, so pjsua-lib does not need to release it manually here. Even better the auto release comes with warning messages in the log so app will be aware of it (and hopefully fix it).

Also, if resource leak is the concern, shouldn't it invoke `pjsua_player/recorder_destroy()` instead of `pj_pool_release()`, so any file descriptors will also be released?